File: examples/rust/constructs/functions_recursion.rs
// Ported from examples/lumen/constructs/functions_recursion.lm by scripts/port_examples.py; edit the Lumen original, not this file.
fn factorial(n: i64) -> i64 {
    if n <= 1 {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

fn countdown(n: i64) -> i64 {
    if n <= 0 {
        println!("Done");
    } else {
        println!("{}", n);
        return countdown(n - 1);
    }
}

fn main() {
    println!("Test: Recursion");
    println!("Factorial of 5:");
    println!("{}", factorial(5));
    println!("Countdown from 3:");
    countdown(3);
}

File: examples/python/constructs/functions_recursion.py
# Ported from examples/lumen/constructs/functions_recursion.lm by scripts/port_examples.py; edit the Lumen original, not this file.
def factorial(n):
    if n <= 1:
        return 1
    else:
        return n * factorial(n - 1)

def countdown(n):
    if n <= 0:
        print("Done")
    else:
        print(n)
        return countdown(n - 1)

print("Test: Recursion")
print("Factorial of 5:")
print(factorial(5))
print("Countdown from 3:")
countdown(3)
